rest-api-spec/src/main/resources/rest-api-spec/test/search/260_sort_mixed.yml

```diff
@@ -0,0 +1,65 @@
+"search across indices with mixed long and double numeric types":
```

Comment:
@gashutos so I find out why test is flipping, this is very interesting. 

The test only works on single node clusters (or, to generalize it, when search results come from single node without serialization being involved). The reason for that is that `SortedNumericSortField`s are deserialized by Apache Lucene as `SortField`s and that is how they come to `SearchPhaseController`.  As you may guessed already, the sort optimization does not trigger anymore because `instanceof SortedNumericSortField` is not met (those are just `SortField` instances).

Reply:
This is concerning @reta , The optimization will still get applied but while merging results, if it has different SortFiled.Type (INT & LONG from two different shards), it will fail the request - 
Let me see what could be done here....

Reply:
> But current implementation works for Int & Long indices. 

No, it does not (Lucene is typecasting), here is the prove:

```
  1> Caused by: java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')
  1>    at java.base/java.lang.Integer.compareTo(Integer.java:71)                                                                                                                             
  1>    at org.apache.lucene.search.FieldComparator.compareValues(FieldComparator.java:115)   
```

Reply:
@gashutos just to may be clarify, this is not the regression caused by optimization changes, it existed before and still exist in Elasticsearch 7.x release line.

Reply:
@reta I verified that this is regression. To reproduce this, do below steps.
1. Ingest http_logs workload from OSB to multi node cluster (I created with 5 nodes)
2. Create and index with exactly same field mapping except ```size``` field as ```long```
3. Re-index any of the http_logs workload index to newly created index on 2nd step.
4. Check this behaviour without code changes in #6424  (it works fine, while with #6424 it breaks).

I raised a PR to fix this behaviour, #6926 lets get that in before this PR.

Reply:
:+1: test case have been updated